lib/ui/semantics.dart

```diff
@@ -641,6 +641,176 @@ class SemanticsFlag {
   }
 }
 
+/// A string that carries a list of [StringAttribute]s.
+class AttributedString extends NativeFieldWrapperClass2 {
+  /// Creates a attributed string.
+  ///
+  /// The [TextRange] in the [attributes] must be inside the length of the
+  /// [string].
+  ///
+  /// The [attributes] must not be changed after the attributed string is
+  /// created.
+  AttributedString(
+    this.string, {
+    this.attributes = const <StringAttribute>[],
+  }) : assert(() {
+         for (final StringAttribute attribute in attributes) {
```

Comment:
Seems like we should be asserting something like this:

```
assert(string.isNotEmpty || attributes.isEmpty)
```

And then I think that resolves my comments for operator +.